File: sample/src/main/java/fr/o80/app/ABTestingActivity.java
package fr.o80.app;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.widget.TextView;

import fr.o80.lib.abtesting.ABTesting;

public class ABTestingActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_ab_testing);

        TextView textView = (TextView) findViewById(R.id.textView);

        // Retrieve ABTest config
        ABTestingApplication application = (ABTestingApplication) getApplication();
        ABTesting abTesting = application.getAbTesting();

        // Apply changes for there current AB testing case
        switch (abTesting.result(ABTestingConst.ABTESTING_EXAMPLE)) {
            case "A":
                textView.setText("(A) case");
                break;
            case "B":
                textView.setText("(B) case");
                break;
            default:
                textView.setText("Default case");
                break;
        }
    }
}

File: sample/src/main/java/fr/o80/app/ABTestingActivity.kt
package fr.o80.app

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_ab_testing.*

class ABTestingActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_ab_testing)

        val app = application as ABTestingApplication

        loadABResult(app)

        resetButton.setOnClickListener {
            app.abTesting.reset(ABTestingConst.ABTESTING_EXAMPLE)
            loadABResult(app)
        }
    }

    fun loadABResult(app: ABTestingApplication) {
        // Apply changes for there current AB testing case
        val abValue = app.abTesting.result(ABTestingConst.ABTESTING_EXAMPLE)
        when (abValue) {
            "A" -> {
                textView.text = "(A) case"
            }
            "B" -> {
                textView.text = "(B) case"
            }
            else -> {
                textView.text = "Default case"
            }
        }
    }
}
